packages/react-reconciler/src/ReactFiberBeginWork.new.js

```diff
@@ -3572,10 +3574,11 @@ function attemptEarlyBailoutIfNoScheduledUpdate(
     case HostRoot:
       pushHostRootContext(workInProgress);
       const root: FiberRoot = workInProgress.stateNode;
+      pushRootTransition(workInProgress, root, renderLanes);
```

Comment:
Nit: the order of pushes in the begin phase needs to be the reverse of the pops in the complete phase.

It happens to not matter here because none of these functions read the current value before the next one pops, but it's a refactor hazard if we ever change them to do that.

So since the complete phase is:

- popRootTransition
- popCacheProvider

The begin phase needs to be:

- pushCacheProvider
- pushRootTransition